apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/interceptor/TransportActionNodeProxyExecuteMethodsInterceptor.java

```diff
@@ -76,7 +76,15 @@ public void handleMethodException(EnhancedInstance objInst, Method method, Objec
 
     @Override
     public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
-        EnhancedInstance actions = (EnhancedInstance) allArguments[2];
+        EnhancedInstance actions;
+        if (allArguments.length > 2) {
+            // (Settings settings, GenericAction<Request, Response> action, TransportService transportService)
+            actions = (EnhancedInstance) allArguments[2];
+        } else {
+            // 7.11 +
+            // (ActionType<Response> action, TransportService transportService)
+            actions = (EnhancedInstance) allArguments[1];
+        }
```

Comment:
This is not a proper way to fix this, please use different interceptors with different instrumentations.

Reply:
fixed